api/v1alpha1/dspipeline_types.go

```diff
@@ -59,69 +56,83 @@ type APIServer struct {
 	Deploy bool `json:"deploy"`
 	// Specify a custom image for DSP API Server.
 	Image string `json:"image,omitempty"`
+	// Create an Openshift Route for this DSP API Server. Default: true
+	// +kubebuilder:default:=true
+	// +kubebuilder:validation:Optional
+	EnableRoute bool `json:"enableOauth"`
+	// Include sample pipelines with the deployment of this DSP API Server. Default: true
+	// +kubebuilder:default:=false
+	// +kubebuilder:validation:Optional
+	EnableSamplePipeline bool   `json:"enableSamplePipeline"`
+	ArgoLauncherImage    string `json:"argoLauncherImage,omitempty"`
+	ArgoDriverImage      string `json:"argoDriverImage,omitempty"`
+	// Specify custom Pod resource requirements for this component.
+	Resources *ResourceRequirements `json:"resources,omitempty"`
+
+	// If the Object store/DB is behind a TLS secured connection that is
+	// unrecognized by the host OpenShift/K8s cluster, then you can
+	// provide a PEM formatted CA bundle to be injected into the DSP
+	// server pod to trust this connection. CA Bundle should be provided
+	// as values within configmaps, mapped to keys.
+	CABundle *CABundle `json:"cABundle,omitempty"`
+
+	// CustomServerConfig is a custom config file that you can provide
+	// for the api server to use instead.
+	CustomServerConfig *ScriptConfigMap `json:"customServerConfigMap,omitempty"`
+
 	// Default: true
+	// Deprecated: v1 only, will be removed in the future.
```

Comment:
hm, should we be bumping our API to v2alpha1 for DSP v2?

Reply:
also, depending on the answer to the above, we may want to say "v1alpha1" here. I think you mean "DSP v1" when you say "v1", but in this CRD API file, "v1" means a kubernetes API and not a version of the project. It could cause confusion.

Reply:
> hm, should we be bumping our API to v2alpha1 for DSP v2?

This is something we should have probably considered way ahead of time. I think crd version conversion strategy is upto us to decide when we do it. I'm not sure what the standard practice is here, but I don't think we should update the version of the crd this close to release, it would be difficult to account for all edge cases we need to handle. 

I updated v1 to say "DSP V1" however.